lib/webhook_json_parser.rb

```diff
@@ -0,0 +1,18 @@
+require 'rack/parser'
+require 'json'
+
+module Sinatra
+  class WebhookJsonParser
+    def call(body)
+      data = JSON.parse(body, quirks_mode: true)
+      branch = (
+        data['ref']                                          ||  # github & gitlab
```

Comment:
Do you have URLs where these references are documented? If so, perhaps we can add the URLs as comments, for any future changes or issues.

Reply:
Why no `rescue nil` here?

Reply:
See comment below - It looks like he just took the code that sets `branch` from the main app file (which I took from the erb in `puppet-r10k`) and placed it here. One of those things that needs to be cleaned up, but it already works so it's further down the list to fix

Reply:
I just lifted the existing code to test out rack-parser.  The current parsing is ugly and there must be a better way. This might not be it though.

Reply:
I just lifted the existing code to test out rack-parser.  The current parsing is ugly and there must be a better way. This might not be it though.